File: Python/0301-0350/344. Reverse String.py
# https://leetcode.com/problems/reverse-string/

class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        start, end = 0, len(s)-1 # Two-Pointers
        temp = ' ' # Temporary Variable 

        while start < end: # Swap the characters in start and end pointers
            temp = s[start]
            s[start] = s[end]
            s[end] = temp
            start += 1
            end -= 1

File: Java/0301-0350/344. Reverse String.java
// https://leetcode.com/problems/reverse-string/

class Solution {
    public void reverseString(char[] s) {
        int start = 0, end = s.length-1; // Two-Pointers
        char temp = ' '; // Temporary Variable 

        while(start < end){ // Swap the characters in start and end pointers
            temp = s[start];
            s[start++] = s[end];
            s[end--] = temp;
        }
    }
}
